aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -109,7 +111,13 @@ public static AppSyncStep builder() {
      * Start subscribing to model mutations.
      */
     synchronized void startSubscriptions() throws DataStoreException {
-        int subscriptionCount = modelProvider.modelNames().size() * SubscriptionType.values().length;
+        List<ModelSchema> modelSchemas = Observable.fromIterable(modelProvider.modelSchemas().values())
+            // Don't subscribe to models that have an associated syncExpression of QueryPredicates.none().
+            .filter(schema -> !QueryPredicates.none().equals(queryPredicateProvider.getPredicate(schema.getName())))
```

Comment:
It's getting to be a long line -- 

```suggestion
            .filter(schema -> {
                String name = schema.getName();
                QueryPredicate predicate = queryPredicateProvider.getPredicate(name);
                return !QueryPredicates.none().equals(predicate);
            })
```